central/globaldb/v2backuprestore/restore/postgres.go

```diff
@@ -0,0 +1,222 @@
+package restore
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"time"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/globaldb"
+	"github.com/stackrox/rox/central/globaldb/v2backuprestore/common"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/retry"
+)
+
+const (
+	// restoreDB - temporary database to apply the postgres dump
+	restoreDB = "central_restore"
+	// createTemplate - template DB to base the temporary DB off of.
+	createTemplate = "template0"
+	// connectDB - database we can connect to in order to perform the rename
+	// TODO: ROX-11272 We should reserve a database for admin type functionality
+	// such as what we are doing with a restore.  This will need to be updated
+	// with that change.
+	connectDB = "template1"
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	postgresOpenRetries        = 10
+	postgresTimeBetweenRetries = 10 * time.Second
+)
+
+// dropDB - drops a database.  This is so we can restore to a new database then flip the name
+// of the restored database to the original for consistency
+func dropDB(sourceMap map[string]string, config *pgxpool.Config, databaseName string) error {
```

Comment:
Can you move postgres databases management like dropDB, createDB, renameDB and some form of admtool under pkg?  And  we need to separate that from the pgxpoo.Config.

Reply:
I think I would like to defer the work with the admin pool to ROX-11272.  I think that would be cleaner to do then we can define an admin DB and an admin source in the central-db config if we desire in addition to a default admin source to use to create an admin pool.  The rest of the stuff I have moved.